### PR TITLE
nagiosPlugins.manubulon-snmp-plugins: init at 2.1.0-unstable-2024-03-13

### LINKED ISSUE #349569: Package request: Manubulon SNMP Plugins
**Project description**

A set of Icinga/Nagios plugins to check hosts and hardware with the SNMP protocol.

**Metadata**

* homepage URL: https://github.com/SteScho/manubulon-snmp
* source URL: https://github.com/SteScho/manubulon-snmp
* license: gpl2Only
* platforms: unix

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by jwillikers**
The [Manubulon SNMP Plugins]( https://github.com/SteScho/manubulon-snmp) are a set of Icinga/Nagios plugins to check hosts and hardware with the SNMP protocol.

Fixes #349569.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by anthonyroussel** [comment]
@ofborg build nagiosPlugins.manubulon-snmp-plugins nagiosPlugins.manubulon-snmp-plugins.tests

File: pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+}:
+stdenv.mkDerivation rec {
+  pname = "manubulon-snmp-plugins";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner = "SteScho";
+    repo = "manubulon-snmp";
+    rev = "v${version}";
+    sha256 = "sha256-I4wKAy704ddoseb7NFBj7+O7l1Sy2JItoe6wrBUaJRQ=";
```
Comment (anthonyroussel) [inline pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix:20]:
```suggestion
    hash = "sha256-I4wKAy704ddoseb7NFBj7+O7l1Sy2JItoe6wrBUaJRQ=";
```

`sha256` attribute is deprecated. `hash` is a replacement.

File: pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    done
+  '';
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.version = testers.testVersion {
+      package = manubulon-snmp-plugins;
+      # Program returns status code 3
+      command = "check_snmp_int.pl --version || true";
+      version = "check_snmp_int version : ${version}";
```
Comment (anthonyroussel) [inline pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix:185]:
```suggestion
```

This line is not needed.
`testers.testVersion` will check that `version` is contained in the output.

File: pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+  src = fetchFromGitHub {
+    owner = "SteScho";
+    repo = "manubulon-snmp";
+    rev = "v${version}";
+    sha256 = "sha256-I4wKAy704ddoseb7NFBj7+O7l1Sy2JItoe6wrBUaJRQ=";
+  };
+
+  patches = [
+    (fetchDebianPatch {
```
Comment (anthonyroussel) [inline pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix:24]: Looks like all these patches are already included upstream.

What do you think about just setting the version directly to an an unstable commit?

```patch
diff --git a/pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix b/pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix
index 7f7f4842768f..f4f206779bac 100644
--- a/pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix
+++ b/pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix
@@ -1,5 +1,4 @@
 {
-  fetchDebianPatch,
   fetchFromGitHub,
   lib,
   makeWrapper,
@@ -11,144 +10,15 @@
 }:
 stdenv.mkDerivation rec {
   pname = "manubulon-snmp-plugins";
-  version = "2.1.0";
+  version = "2.1.0-unstable-2024-03-13";
 
   src = fetchFromGitHub {
     owner = "SteScho";
     repo = "manubulon-snmp";
-    rev = "v${version}";
-    sha256 = "sha256-I4wKAy704ddoseb7NFBj7+O7l1Sy2JItoe6wrBUaJRQ=";
+    rev = "1a5afb2486802034146277010d956eba9c2ad54b";
+    sha256 = "sha256-B5CCGMkNv1wGnLQIl0yiGTH2j5MOlj5+MrRqbLNIwhE=";
   };
 
-  patches = [
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "10_check_snmp_storage_error_handling";
-      hash = "sha256-P8srMAAcYiSJYww7eBRN78aCSQPq2+fqOJqJ187l2u8=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "11_check_snmp_int_agent_workaround";
-      hash = "sha256-BXC9QwB/3ztpJCTBZqeCfTZYY5DPwyPJjy62KQqiEcQ=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "12_check_snmp_mem_perf";
-      hash = "sha256-cAH/Oj/r35xMKn56w3jXOtoCx1Wfe2YWTCOS6PFkiTQ=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "13_check_snmp_process_new_features";
-      hash = "sha256-nqZEy4UFtz3o6taJ7zNhHWjQXtsavEdVuhdUH6kfLSM=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "14_check_snmp_int_ign";
-      hash = "sha256-ODOsLu7avqmEmmt6EC4ZlPn6bXeEqVvaUZxOpFnzx98=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "15_check_snmp_int_spaces";
-      hash = "sha256-WvA0/DasyMXJgkNhNQNFYatZ4rT32ybbreomIE2z0vc=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "16_check_snmp_win_max_message_size";
-      hash = "sha256-f28TokPhukMqmNRSpA0ebMHPHYO6Xnq7ABTb19dNtw0=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "17_check_snmp_storage_okifempty";
-      hash = "sha256-OZ+0y65rDekD3G21vXmJsWltMyfx2+vQ8wUGdLgd2D0=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "18_check_snmp_int_fix_perf_out";
-      hash = "sha256-1jsDCTCUqYPIJNzlWMVJeUHEZ5tB/NhNex2oLv8dc5U=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      patch = "19_check_snmp_int_remove_unneeded_my";
-      debianRevision = "4";
-      hash = "sha256-C3Bh94aRRq94kg94un3rEaCb5W7z3gNiD4n8FFel9fk=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      patch = "20_check_snmp_int_avaid_huge_amount_of_regex";
-      debianRevision = "4";
-      hash = "sha256-Kh3hk7xZNWHmYlDM4k6QTFdetklspsBVvIl2S9SO6AA=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "21_check_snmp_load_update_fortiswitch_and_fortigate4.3";
-      hash = "sha256-D6q1ZZ0tMHcvIu8EWyx4A5w+5Tzj68JK30kL7fm679A=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "22_check_snmp_storage_fix_space_btrfs";
-      hash = "sha256-fqq1UUmqzYdxEpcrelGxrhqRghaH7BWAueTBmr8czW0=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "23_check_snmp_int_interface_speed";
-      hash = "sha256-dn5yzqsSfwpVimxzBKIEhX64RR0C29ORiv3F+uKUIk8=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "24_tcp_udp_ipv4_ipv6";
-      hash = "sha256-W5bEGqX2bAT4t0jey6p9TBT28ckiLPxPZWcajjyI/xU=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "25_check_snmp_int_add_metrik";
-      hash = "sha256-5IlnjOAb9e2SscG5fBQ7hfJXAOwZtKOGWnPuQigXrUo=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "50_disable_epn";
-      hash = "sha256-Qsf/4JeLV2P+XUnHDCH6QYaRL7bdFJQSEdhpkfsZTgw=";
-      pname = "nagios-snmp-plugins";
-    })
-    (fetchDebianPatch {
-      inherit version;
-      debianRevision = "4";
-      patch = "51_fix_privacy_doc";
-      hash = "sha256-ZIYrNCZSTWSfgcR+Oe5Nuap/az3DjuD+Uil05RtXE+M=";
-      pname = "nagios-snmp-plugins";
-    })
-  ];
-
   buildInputs = with perlPackages; [
     CryptDES
     CryptRijndael
```

Comment (jwillikers) [reply pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix:24]: That sounds like a fine idea to me.

File: pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix (hunk 1 above)
Comment (jwillikers) [reply pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix]: Done.

File: pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix (hunk 2 above)
Comment (jwillikers) [reply pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix]: Done, thanks!

File: pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix (hunk 3 above)
Comment (jwillikers) [reply pkgs/servers/monitoring/nagios-plugins/manubulon-snmp-plugins/default.nix]: Done.

**Review by anthonyroussel — APPROVED** [review]

**Comment by anthonyroussel** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349583`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>nagiosPlugins.manubulon-snmp-plugins</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>nagiosPlugins.manubulon-snmp-plugins</li>
  </ul>
</details>

**Pull request merged**